sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/services/OrchestrationService.java

```diff
@@ -87,35 +88,69 @@ public OrchestrationChatResponse messagesHistory(@Nonnull final String prevMessa
   }
 
   /**
-   * Apply both input and output filtering for a request to orchestration.
+   * Apply input filtering for a request to orchestration.
    *
    * @link <a
    *     href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/input-filtering">SAP
    *     AI Core: Orchestration - Input Filtering</a>
+   * @param policy the explicitness of content that should be allowed through the filter
+   * @return the assistant response object
+   */
+  @Nonnull
+  public OrchestrationChatResponse inputFiltering(@Nonnull final AzureFilterThreshold policy) {
+    final var prompt =
+        new OrchestrationPrompt("'We shall spill blood tonight', said the operation in-charge.");
+    final var filterConfig =
+        new AzureContentFilter().hate(policy).selfHarm(policy).sexual(policy).violence(policy);
+
+    final var configWithFilter = config.withInputFiltering(filterConfig);
+
+    try {
+      return client.chatCompletion(prompt, configWithFilter);
+    } catch (OrchestrationClientException e) {
+      if (e.getMessage().contains("400 Bad Request")) {
+        log.info("Content filtered out by input filter before any response was available.");
+      }
+      throw e;
+    }
```

Comment:
(Question)

What is the reason to catch the exception and log conditionally to `info` while rethrowing the same exception again?

```suggestion
    return client.chatCompletion(prompt, configWithFilter);
```

Reply:
I am trying to balance e2e testing (why the rethrow) and demonstration of handling response object on input filtering.

I see similar approach here [code from JavaScript](https://github.com/SAP/ai-sdk-js/blob/main/sample-code/src/orchestration.ts#L87).

Reply:
Nevermind, I see the problem.